crates/prelude/assets/extensions/gd.php

```diff
@@ -0,0 +1,697 @@
+{
+}
+
+function imageloadfont(string $filename): \GdFont|false
+{
+}
+
+function imagesetstyle(\GdImage $image, array $style): bool
+{
+}
+
+function imagecreatetruecolor(int $width, int $height): \GdImage|false
+{
+}
+
+function imageistruecolor(\GdImage $image): bool
+{
+}
+
+function imagetruecolortopalette(\GdImage $image, bool $dither, int $num_colors): bool
+{
+}
+
+function imagepalettetotruecolor(\GdImage $image): bool
+{
+}
+
+function imagecolormatch(\GdImage $image1, \GdImage $image2): bool
+{
+}
+
+function imagesetthickness(\GdImage $image, int $thickness): bool
+{
+}
+
+function imagefilledellipse(\GdImage $image, int $center_x, int $center_y, int $width, int $height, int $color): bool
+{
+}
+
+function imagefilledarc(\GdImage $image, int $center_x, int $center_y, int $width, int $height, int $start_angle, int $end_angle, int $color, int $style): bool
+{
+}
+
+function imagealphablending(\GdImage $image, bool $enable): bool
+{
+}
+
+function imagesavealpha(\GdImage $image, bool $enable): bool
+{
+}
+
+function imagelayereffect(\GdImage $image, int $effect): bool
+{
+}
+
+function imagecolorallocatealpha(\GdImage $image, int $red, int $green, int $blue, int $alpha): int|false
+{
+}
+
+function imagecolorresolvealpha(\GdImage $image, int $red, int $green, int $blue, int $alpha): int
+{
+}
+
+function imagecolorclosestalpha(\GdImage $image, int $red, int $green, int $blue, int $alpha): int
+{
+}
+
+function imagecolorexactalpha(\GdImage $image, int $red, int $green, int $blue, int $alpha): int
+{
+}
+
+function imagecopyresampled(\GdImage $dst_image, \GdImage $src_image, int $dst_x, int $dst_y, int $src_x, int $src_y, int $dst_width, int $dst_height, int $src_width, int $src_height): bool
+{
+}
+
+function imagerotate(\GdImage $image, float $angle, int $background_color): \GdImage|false
+{
+}
+
+function imagesettile(\GdImage $image, \GdImage $tile): bool
+{
+}
+
+function imagesetbrush(\GdImage $image, \GdImage $brush): bool
+{
+}
+
+function imagecreate(int $width, int $height): \GdImage|false
+{
+}
+
+function imagetypes(): int
+{
+}
+
+function imagecreatefromstring(string $data): \GdImage|false
+{
+}
+
+function imagecreatefromavif(string $filename): \GdImage|false
+{
+}
+
+function imagecreatefromgif(string $filename): \GdImage|false
+{
+}
+
+function imagecreatefromjpeg(string $filename): \GdImage|false
+{
+}
+
+function imagecreatefrompng(string $filename): \GdImage|false
+{
+}
+
+function imagecreatefromwebp(string $filename): \GdImage|false
+{
+}
+
+function imagecreatefromxbm(string $filename): \GdImage|false
+{
+}
+
+function imagecreatefromwbmp(string $filename): \GdImage|false
+{
+}
+
+function imagecreatefromgd(string $filename): \GdImage|false
+{
+}
+
+function imagecreatefromgd2(string $filename): \GdImage|false
+{
+}
+
+function imagecreatefromgd2part(string $filename, int $x, int $y, int $width, int $height): \GdImage|false
+{
+}
+
+function imagecreatefrombmp(string $filename): \GdImage|false
```

Comment:
should we add `@psalm-ignore-falsable-return`? It is supported by Mago.

This makes it so Mago does not complain about `false` when using the result as a `GdImage`